### ISSUE TITLE
Optimize ISA-Tab import performance

### ISSUE THREAD

**Issue by hackdna** [parklab/refinery-platform#463]
### Steps to reproduce
Import http://stemcellcommons.org/sites/default/files/isa/isa_14911_906861.zip via /data_set_manager/import/isa-tab-form/

### Observed results
Import took about an hour in Vagrant VM with system load close to 3 and Celery worker process consuming over 700MB of memory. The VM eventually ran out of memory completely and terminated the Celery worker which resulted in the following error message:
```
Traceback:
File "/home/vagrant/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/django/core/handlers/base.py" in get_response
  115.                         response = callback(request, *callback_args, **callback_kwargs)
File "/home/vagrant/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/django/contrib/auth/decorators.py" in _wrapped_view
  25.                 return view_func(request, *args, **kwargs)
File "/home/vagrant/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/django/views/generic/base.py" in view
  68.             return self.dispatch(request, *args, **kwargs)
File "/home/vagrant/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/django/views/generic/base.py" in dispatch
  86.         return handler(request, *args, **kwargs)
File "/vagrant/refinery/data_set_manager/views.py" in post
  236.                                               False, temp_file_path).get()
File "/home/vagrant/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/celery/result.py" in get
  95.                                                    interval=interval)
File "/home/vagrant/.virtualenvs/refinery-platform/local/lib/python2.7/site-packages/celery/backends/base.py" in wait_for
  147.                     raise result

Exception Type: WorkerLostError at /data_set_manager/import/isa-tab-form/
Exception Value: Worker exited prematurely.
```
On a server with 12 cores and 16GB of memory, importing this dataset resulted in the following error shown in the browser and no dataset was created:
```
504 Gateway Time-out
The gateway did not receive a timely response from the upstream server or application.
```
### Expected results
1. Metadata import process should not take that many resources and run for more than a minute.
2. WorkerLostError exception should be handled with an appropriate message provided to the user.

**Comment by flekschas** [comment]
It works for me when importing it via the command line. It's not very fast though.

**Comment by hackdna** [comment]
Server VM just ran out of memory while importing isa_14911_906861.zip. Celery worker that was doing the import was using 85% of ~16GB total just before the crash.

**Comment by ngehlenborg** [comment]
@flekschas: Do you still have this data set imported? Could you check out what the provenance graph looks like and post a screenshot of the whole thing here?

**Comment by hackdna** [comment]
I was trying to share this data set using `make_data_set_public` command. The process ran for several minutes and was taking ~12GB of memory while running `search_indexes prepare`. It finally crashed during `pysolr add`.

**Comment by ngehlenborg** [comment]
@hackdna: so you were able to import the data set in the end?

**Comment by flekschas** [comment]
Wow! Maybe the `text` property gets so massive that it runs out of memory. I noticed that some annotated notes are duplicated 40 times, blowing up the description text massively. I'll investigate.

Just for quick reference, the title is:

> Genome-wide location analysis of WNT (Tcf7l2) and BMP (SMAD1) in human hematopoeitic cell lines co-occupied with lineage specific regulators (GATA1, GATA2, CEBPA)

**Comment by hackdna** [comment]
Yes, the worker crashed during import but a DataSet model instance was created.

**Comment by flekschas** [comment]
Might be related to #568.

**Comment by ngehlenborg** [comment]
The issue with this file is incorrect annotation. For example, the data transformation `alignment` in the assay file has an empty output - no output filenames are provided - which leads to a combinatorial explosion of possible paths in the experiment graph. There are other issues, such as a data transformation that outputs PDF files (`MACS2_TF_model`) feeding into a data transformation (`MACS2_TF_signal`) that produces bigwig files.

**Comment by hackdna** [comment]
Related to #842 and #843.

**Comment by hackdna** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;uPviGMWp&#x2F;199-improve-data-model-of-data-sets"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> Improve data model of data sets</a>
